src/DistributedTriangulations.jl

```diff
@@ -1,4 +1,52 @@
 
+function filter_cells_when_needed(strategy::AssemblyStrategy, model::DiscreteModel, trian::Triangulation)
+  @abstractmethod
+end
+
+function filter_cells_when_needed(strategy::RowsComputedLocally, model::DiscreteModel, trian::Triangulation)
+  trian
+end
+
+function filter_cells_when_needed(strategy::OwnedCellsStrategy, model::DiscreteModel, trian::Triangulation)
+  remove_ghost_cells(trian,strategy.part,strategy.cell_gids)
+end
+
+function filter_cells_when_needed(strategy::OwnedCellsStrategy, model::DiscreteModel, trian::SkeletonTriangulation)
+  topo = get_grid_topology(model)
+  D = num_cell_dims(model)
+  facets_to_cells = Gridap.Geometry.get_faces(topo,D-1,D)
```

Comment:
I think this should be implemented as a new method of  `remove_ghost_cells` specialized for `SkeletonTriangulation` in order to promote code re-usage.

Moreover, perhaps, the implementation can be simplified a bit with something like this (to ckech that it works):

```julia
function remove_ghost_cells(trian::SkeletonTriangulation,part,cell_gids)
  left = remove_ghost_cells(trian.left,part,cell_gids)
  right = reindex(trian.right,left.cell_to_oldcell)
SkeletonTriangulation(left,right)
end
```

Reply:
> I think this should be implemented as a new method of remove_ghost_cells specialized for SkeletonTriangulation in order to promote code re-usage.

Done! The old method (interface & rationale) resulted from the fact that I did not realize that I can extract the cells sharing a facet already from `SkeletonTriangulation`, i.e., I do not need the topology data provided by `DiscreteModel`.

> Moreover, perhaps, the implementation can be simplified a bit with something like this (to ckech that it works):

If I am not missing something, I think the implementation you proposed is not correct (I did not check it). In particular, it leads to an "inconsistent" output `SkeletonTriangulation` (lets call it `output_trian`)  in the sense that it does NOT fullfill the following assertion: `num_cells(output_trian.left) == num_cells(output_trian.right)`. See also the attached scanned diagram to clarify this point. 

[CamScanner 06-03-2020 12.31.32.pdf](https://github.com/gridap/GridapDistributed.jl/files/4720919/CamScanner.06-03-2020.12.31.32.pdf)



Do u agree or am I totally confused?

Finally, I am wondering if `remove_ghost_cells` is the appropiate name for the function at hand. Perhaps something like: `build_part_local_integration_triangulation` (or something in this line) would be more intention-revealing. What do u think?

Reply:
> it leads to an "inconsistent" output SkeletonTriangulation (lets call it output_trian) in the sense that it does NOT fullfill the following assertion: num_cells(output_trian.left) == num_cells(output_trian.right).

I would say that this assertion is indeed fulfilled. Anyway, I am not sure if my proposed approach is consistent across subdomains. So, let's take your approach.

Reply:
> I would say that this assertion is indeed fulfilled.

You are right. It is fulfilled. It is guaranteed by the fact that we re-use `left.cell_to_old_cell` to create the portion of the "right" boundary triangulation of the original skeleton triangulation.
I missed that sentence, and (wrongly) though that it was actually ` left = remove_ghost_cells(trian.right,part,cell_gids)` (my diagram correspond to this latter scenario).

> Anyway, I am not sure if my proposed approach is consistent across subdomains.

I see what you mean. Your approach works as far as for any facet shared among two subdomains, the concept of "left" and "right" cell matches in both subdomains. If it doesnt for a given facet, you end up adding the contribution of that facet twice to the global linear system. If am not wrong, at present, at a given processor, the left cell is the one with smallest LOCAL cell identifier among the two. As, in general, a processor is free to select the local identifier of its cells, this is highly dangerous. Thus, my approach ensures that the problem is solved indepently of how the local cells are numbered.

Reply:
BTW, another (minor?) consequence (minor drawback?) of your approach vs mine, is that yours requires `left` to be a `TriangulationPortion`, otherwise duck typing fails, while mine does not.